assignments/array-methods.js

```diff
@@ -55,23 +55,27 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
+let fullName = []; 
+runners.forEach(runner => fullName.push(runner.first_name, runner.last_name));
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+// let allCaps = [];
+const map1 = runners.map(name => name.first_name.toUpperCase());
+
+console.log(map1); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.find(runner => runner.shirt_size === 'L');
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+let donations = runners.map(runner => runner.donation);
+let total = donations.reduce((a,b) => a + b );
+console.log(total);
 
```

Comment:
I remember going through this over a meeting. This is nice. Another way to make this work is to "chain" map and reduce or only using "reduce". This works too.